File: test_vectors/python/write_out.py
#!/usr/bin/env python3
import sys

def main():
    out = sys.argv[1] if len(sys.argv) > 1 else 'out.txt'
    with open(out, 'w', encoding='utf-8', newline='\n') as f:
        f.write('hello\n')

if __name__ == '__main__':
    main()

File: test_vectors/node/write_out.js
#!/usr/bin/env node
/**
 * STUNIR Test Vector - Node.js File Writer
 *
 * Simple utility to write test output files for Node.js compatibility testing.
 * Part of the STUNIR test vectors for verifying Node.js toolchain integration.
 *
 * @module write_out
 * @example
 *   node write_out.js output.txt
 */

const fs = require('fs');

/**
 * Output file path from command line arguments.
 * Defaults to 'out.txt' if not provided.
 * @type {string}
 */
const out = process.argv[2] || 'out.txt';

/**
 * Write test content to the specified file.
 *
 * @param {string} filepath - Path to the output file
 * @returns {void}
 */
function writeOutput(filepath) {
    fs.writeFileSync(filepath, 'hello\n', { encoding: 'utf8' });
}

// Execute the write operation
writeOutput(out);
